bin/cheribsdtest/cheribsdtest_vm.c

```diff
@@ -123,6 +123,32 @@ CHERIBSDTEST(cheribsdtest_vm_tag_shm_open_anon_private,
 	cheribsdtest_success();
 }
 
+CHERIBSDTEST(cheribsdtest_vm_tag_mprotect_none_and_back,
+    "check that mmap/mprotect(NONE)/mprotect(RW) still permits stores")
+{
+	size_t sz = getpagesize();
+	void * __capability volatile * arena;
+
+	arena = CHERIBSDTEST_CHECK_SYSCALL(mmap(NULL, sz, PROT_READ|PROT_WRITE,
+	    MAP_ANON, -1, 0));
+
+	// store cap, implicitly testing store permission
+	arena[0] = __DEVOLATILE(void * __capability, arena);
```

Comment:
`&arena` avoids the need to devolatile, though neither of these are going to work in hybrid